android/src/main/java/com/apparence/camerawesome/CamerawesomePlugin.java

```diff
@@ -483,18 +515,34 @@ private void _handleRecordVideo(final MethodCall call, final Result result) {
             result.error("PATH_NOT_SET", "a file path must be set", "");
             return;
         }
-        String path = call.argument("path");
-        try {
-            mCameraPicture.recordVideo(
-                    mCameraStateManager.getCameraDevice(),
-                    path,
-                    mCameraSetup.getOrientation()
-            );
+        final String path = call.argument("path");
 
-            result.success(null);
-        } catch (CameraAccessException | IOException e) {
-            result.error(e.getMessage(), "cannot open camera", "");
-        }
+        backgroundHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    mCameraPicture.recordVideo(
+                            mCameraStateManager.getCameraDevice(),
+                            path,
+                            mCameraSetup.getOrientation(getOrientationArgument(call))
+                    );
+                } catch (CameraAccessException | IOException e) {
```

Comment:
should we also catch Exception in another catch block and return an unknown exception?

Reply:
Those unchecked exceptions are catched by the flutter method channel and communicated back to the user of this lib as a PlatformException so it feels a bit too much to also handle that case here.